azurecompute/src/main/java/org/jclouds/azurecompute/compute/AzureComputeServiceAdapter.java

```diff
@@ -127,20 +131,61 @@
-              .os(os)
-              .username(loginUser)
-              .password(loginPassword)
-              .sourceImageName(OSImageToImage.fromGeoName(template.getImage().getId())[0])
-              .mediaLink(createMediaLink(storageAccountName, name))
-              .size(RoleSize.Type.fromString(template.getHardware().getName()))
-              .externalEndpoints(externalEndpoints)
-              .subnetName(subnetName)
-              .virtualNetworkName(virtualNetworkName)
-              .reservedIPName(reservedIPAddress)
-              .build();
-
+      final DeploymentParams params;
+
+      if (os == OSImage.Type.LINUX) {
+         LinuxConfigurationSetParams linuxConfig = LinuxConfigurationSetParams.builder().userName(loginUser)
+               .userPassword(loginPassword).build();
+
+         final OSVirtualHardDiskParam osParam = OSVirtualHardDiskParam.builder()
+               .sourceImageName(OSImageToImage.fromGeoName(template.getImage().getId())[0])
+               .mediaLink(createMediaLink(storageAccountName, name))
+               .os(os).diskName(name)
+               .diskLabel(name)
+               .build();
+
+         final RoleParam roleParam = RoleParam.builder()
+               .roleName(name)
+               .roleSize(RoleSize.Type.fromString(template.getHardware().getName()))
+               .osVirtualHardDiskParam(osParam)
+               .linuxConfigurationSet(linuxConfig)
+               .build();
+
+         params = DeploymentParams.builder()
+               .name(name)
+               .roleParam(roleParam)
+               .externalEndpoints(externalEndpoints)
+               .subnetName(subnetName)
+               .virtualNetworkName(virtualNetworkName)
+               .reservedIpName(reservedIPAddress)
+               .build();
+      } else {
+         WindowsConfigurationSetParams winConfig = WindowsConfigurationSetParams.builder().adminUserName(loginUser)
+               .adminPassword(loginPassword).build();
+
+         final OSVirtualHardDiskParam osParam = OSVirtualHardDiskParam.builder()
+               .sourceImageName(OSImageToImage.fromGeoName(template.getImage().getId())[0])
+               .mediaLink(createMediaLink(storageAccountName, name))
+               .os(os)
+               .diskName(name)
+               .diskLabel(name)
+               .build();
+
+         final RoleParam roleParam = RoleParam.builder()
+               .roleName(name)
+               .roleSize(RoleSize.Type.fromString(template.getHardware().getName()))
+               .osVirtualHardDiskParam(osParam)
+               .windowsConfigurationSet(winConfig)
+               .build();
+
+         params = DeploymentParams.builder()
+               .name(name).roleParam(roleParam)
+               .virtualNetworkName(virtualNetworkName)
+               .externalEndpoints(externalEndpoints)
+               .subnetName(subnetName)
+               .reservedIpName(reservedIPAddress)
+               .build();
+      }
```

Comment:
All code here is duplicated. Extract it all out of the conditionals and put only the configuration set in it. You're using builders, so there's no problem on doing that.